### PR TITLE
US92183 - Move polymer dependency and general cleanup

### PR DESCRIPTION
**Pull request by svanherk**
If we move the polymer dependency to a dev dependency (as it's currently only used in the tests), this can skip hybrid and jump to Polymer 2.

### CODE REVIEW

**Comment by dlockhart** [comment]
Hmm, yeah I guess there's really nothing going on in the behaviour itself (or any behaviour) that's Polymer-related, crazy!

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "postinstall": "bower install",
```
Comment (dbatiste) [inline package.json]: I think we will want: `"postinstall": "polymer install --variants",`

File: bower.json [hunk 2]
```diff
@@ -19,18 +19,17 @@
...
     ".editorconfig",
-    ".eslintrc",
+    ".eslintrc.json",
     ".gitignore",
     ".travis.yml",
     "package.json",
+    "wct.conf.json",
     "test",
     "tests",
     "/demo/"
```
Comment (dbatiste) [inline bower.json]: I think this can just be `demo`, and I think we should only need `test` or `tests`.  Also, we can add `polymer.json` to this list.

Comment (dbatiste) [reply bower.json]: Oh, `bower_components` and `node_modules` doesn't make any sense for bower ignore list.

**Comment by dbatiste** [comment]
> Hmm, yeah I guess there's really nothing going on in the behaviour itself (or any behaviour) that's Polymer-related, crazy!

I was thinking about this a little...  Something to watch out for with behaviors is whether or not it should actually be a behavior vs. a shared library.  As a litmus test, does the "behavior" add or change properties/attributes, add listeners or raise events, need access to DOM? Are the methods stateful?  If no, then it starts to seem more like a shared library than a behavior.  There is definitely extra logic that's run to mix behaviors with components, and while true component behaviors can add lots of value, I'm more of a fan of keeping things in a stateless lib where its possible, since it's easier to test, more predictable, etc.

Somewhat related: we have the [d2l-polymer-behaviors-ui](https://github.com/Brightspace/d2l-polymer-behaviors-ui) repo, which in spite of its name, actually only contains shared libraries.  I think we just named it that way since we weren't sure how the cards were going to fall in terms of sharing logic between polymer components.  While much of, if not all of the logic does not rely on Polymer, the unit tests use Polymer web components as fixtures.  Perhaps it would be better to use vanilla components for those fixtures.

Anyway, food for thought.  ;)

**Comment by ryantmer** [comment]
@dbatiste Yeah this repo as well as `d2l-hm-constants-behavior` are kind of in the same boat - made as Polymer behaviours as we weren't really sure how everything was going to fit together in the long run, or if they'd need to be behaviour-y at some point. I think by this point we can fairly confidently say no, but 🤷‍♂️ 

(completely unrelated side note - why are the shrug emoji on github specifically gendered? why... why that.)

**Pull request merged**